SP: Linear Matrix Test Fail

On conda-forge with ImpactX 26.04 we see the `tests/python/test_lattice_optics.py::test_lattice_linear_map` test fail in single precision:
```
 │ FAILED tests/python/test_lattice_optics.py::test_lattice_linear_map - assert False
 │  +  where False = <function allclose at 0x7f45de367830>(array([[ 3.7467057e-01,  2.5400584e+00,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00, -2.3486473e-01],\n       [-4.7621906e-01, -5.5948943e-01,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00,  3.2064538e-02],\n       [ 0.0000000e+00,  0.0000000e+00,  1.6487212e+00,  6.5948849e+00,\n         0.0000000e+00,  0.0000000e+00],\n       [ 0.0000000e+00,  0.0000000e+00,  5.2109534e-01,  2.6909118e+00,\n         0.0000000e+00,  0.0000000e+00],\n       [ 9.9833429e-02,  4.9958169e-02,  0.0000000e+00,  0.0000000e+00,\n         1.0000000e+00, -1.6647103e-03],\n       [ 0.0000000e+00,  0.0000000e+00,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00,  1.0000000e+00]], dtype=float32), array([[ 3.74670546e-01,  2.54005827e+00,  0.00000000e+00,\n         0.00000000e+00,  0.00000000e+00, -2.34864805e-01],\n       [-4.76219076e-01, -5.59489407e-01,  0.00000000e+00,\n         0.00000000e+00,  0.00000000e+00,  3.20646253e-02],\n       [ 0.00000000e+00,  0.00000000e+00,  1.64872127e+00,\n         6.59488508e+00,  0.00000000e+00,  0.00000000e+00],\n       [ 0.00000000e+00,  0.00000000e+00,  5.21095305e-01,\n         2.69091188e+00,  0.00000000e+00,  0.00000000e+00],\n       [ 9.98334297e-02,  4.99583537e-02,  0.00000000e+00,\n         0.00000000e+00,  1.00000000e+00, -1.66466013e-03],\n       [ 0.00000000e+00,  0.00000000e+00,  0.00000000e+00,\n         0.00000000e+00,  0.00000000e+00,  1.00000000e+00]]))
 │  +    where <function allclose at 0x7f45de367830> = np.allclose
 │  +    and   array([[ 3.7467057e-01,  2.5400584e+00,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00, -2.3486473e-01],\n       [-4.7621906e-01, -5.5948943e-01,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00,  3.2064538e-02],\n       [ 0.0000000e+00,  0.0000000e+00,  1.6487212e+00,  6.5948849e+00,\n         0.0000000e+00,  0.0000000e+00],\n       [ 0.0000000e+00,  0.0000000e+00,  5.2109534e-01,  2.6909118e+00,\n         0.0000000e+00,  0.0000000e+00],\n       [ 9.9833429e-02,  4.9958169e-02,  0.0000000e+00,  0.0000000e+00,\n         1.0000000e+00, -1.6647103e-03],\n       [ 0.0000000e+00,  0.0000000e+00,  0.0000000e+00,  0.0000000e+00,\n         0.0000000e+00,  1.0000000e+00]], dtype=float32) = smallmatrix_to_numpy()
 │  +      where smallmatrix_to_numpy = SmallMatrix([[0.374671, 2.54006, 0, 0, 0, -0.234865],\n             [-0.476219, -0.559489, 0, 0, 0, 0.0320645],\n             [0, 0, 1.64872, 6.59488, 0, 0],\n             [0, 0, 0.521095, 2.69091, 0, 0],\n             [0.0998334, 0.0499582, 0, 0, 1, -0.00166471],\n             [0, 0, 0, 0, 0, 1]]).to_numpy
 │ ============ 1 failed, 64 passed, 1 skipped, 2 deselected in 12.05s ============
```

--- a/tests/python/test_lattice_optics.py
+++ b/tests/python/test_lattice_optics.py
@@ -10,7 +10,7 @@
 import numpy as np
 import pytest
 
-from impactx import RefPart, elements
+from impactx import Config, RefPart, elements
 
 
 def test_lattice_linear_map():
@@ -43,9 +43,16 @@ def test_lattice_linear_map():
         ]
     )
 
+    if Config.precision == "SINGLE":
+        atol = 1.0e-7
+        rtol = 5.0e-5
+    else:
+        atol = 0.0
+        rtol = 1.0e-8
+
     # Calculate Linear Transfer Map
     R = lattice.transfer_map(ref)
-    assert np.allclose(R.to_numpy(), R_expected)
+    assert np.allclose(R.to_numpy(), R_expected, rtol=rtol, atol=atol)
 
     # Check unexpected/unsupported options
     with pytest.raises(RuntimeError):
@@ -60,4 +67,4 @@ def test_lattice_linear_map():
 
     # Now the user explicitly assumes that undefined maps are identity maps
     R = lattice.transfer_map(ref, fallback_identity_map=True)
-    assert np.allclose(R.to_numpy(), R_expected)
+    assert np.allclose(R.to_numpy(), R_expected, rtol=rtol, atol=atol)

